native/c/zds.cpp

```diff
@@ -178,23 +178,24 @@ int zds_write_to_dsn(ZDS *zds, std::string dsn, std::string &data)
   }
 
   const string dsname = "//'" + dsn + "'";
-
   std::string temp = data;
-  if (!data.empty())
+
+  auto *fp = fopen(dsname.c_str(), zds->encoding_opts.data_type == eDataTypeBinary ? "wb,recfm=U" : "w");
+  if (nullptr == fp)
   {
-    auto *fp = fopen(dsname.c_str(), zds->encoding_opts.data_type == eDataTypeBinary ? "wb,recfm=U" : "w");
-    if (fp == nullptr)
-    {
-      zds->diag.e_msg_len = sprintf(zds->diag.e_msg, "Could not open '%s'", dsname.c_str());
-      return RTNCD_FAILURE;
-    }
+    zds->diag.e_msg_len = sprintf(zds->diag.e_msg, "Could not open '%s'", dsname.c_str());
+    return RTNCD_FAILURE;
+  }
 
+  if (!data.empty())
+  {
     if (hasEncoding)
     {
       try
       {
-        const auto bytes_with_encoding = zut_encode(temp, "UTF-8", codepage, zds->diag);
-        temp = bytes_with_encoding;
+        // const auto bytes_with_encoding = zut_encode(temp, "UTF-8", codepage, zds->diag);
+        // temp = bytes_with_encoding;
+        temp = zut_encode(temp, "UTF-8", codepage, zds->diag);
```

Comment:
We can probably remove these commented lines, thanks for cleaning up this block

Reply:
resolved in e420726